app/main/lib/shared_models/video_model.py

```diff
@@ -110,19 +110,35 @@ def search_by_context(self, context):
             db.session.rollback()
             raise e
 
+    def get_blocked_response(self, task):
+        callback_url =  Presto.add_item_callback_url(app.config['ALEGRE_HOST'], "video")
+        if task.get("doc_id") is    None:
+            task["doc_id"] = str(uuid.uuid4())
+        response = json.loads(Presto.send_request(app.config['PRESTO_HOST'], "video__Model", callback_url, task, False).text)
+        # Warning: this is a blocking hold to wait until we get a response in 
+        # a redis key that we've received something from presto.
+        return Presto.blocked_response(response, "video")
+
+    def download_temp_file(self, task):
+        result = self.get_blocked_response(task).get("body")
```

Comment:
Per the comment below, I think `download_temp_file` should be having the `results` object from Presto as input and not be calling `get_blocked_response`, which calls Presto another time. At the time this function is called, we already have Presto output and should be using that rather than calling Presto a second time.